src/main/java/tictactoe/PlayerFactory.java

```diff
@@ -3,18 +3,22 @@
 public class PlayerFactory {
     private final Display display;
     private final InputValidator inputValidator;
+    private final Minimax minimax;
 
     public PlayerFactory
-            (Display display, InputValidator inputValidator) {
+            (Display display, InputValidator inputValidator, Minimax minimax) {
         this.display = display;
         this.inputValidator = inputValidator;
+        this.minimax = minimax;
     }
 
     public Player createNewPlayer(String playerType, String mark) {
```

Comment:
I like that you are keeping the logic for each of your players separate, although I'm not sure you need to pass an instance of `Minimax` into the factory. Have you come across the **Template** or **Strategy** design patterns? I think either of these might be useful for you in this situation.

Reply:
Thanks, Ashley. I haven't, but I will look into it this week